src/solc_0.7/GameToken/GameToken.sol

```diff
@@ -0,0 +1,419 @@
+        bytes calldata /*data*/
+    ) external view override returns (bytes4) {
+        if (operator == address(this)) {
+            return ERC1155_RECEIVED;
+        }
+        revert("ERC1155_REJECTED");
+    }
+
+    /// @notice Return the name of the token contract
+    /// @return The name of the token contract
+    function name() external pure override returns (string memory) {
+        return "The Sandbox: GAME token";
+    }
+
+    /// @notice Function to get the symbol of the token contract
+    /// @return The symbol of the token contract
+    function symbol() external pure override returns (string memory) {
+        return "GAME";
+    }
+
+    /// @notice Set the URI of a specific game token
+    /// @param gameId The id of the game token
+    /// @param URI The URI string for the token's metadata
+    function setTokenURI(uint256 gameId, string calldata URI) external override {
+        require(msg.sender == _ownerOf(gameId) || _gameEditors[gameId][msg.sender], "URI_ACCESS_DENIED");
+        _setTokenURI(gameId, URI);
+    }
+
+    /// @notice Function to burn a GAME token and recover assets
+    /// @param from The address of the one destroying the game
+    /// @param to The address to send all GAME assets to
+    /// @param gameId The id of the GAME to destroy
+    /// @param assetIds The assets to recover from the burnt GAME
+    /// @param values The amount of each asset to recover
+    function destroyAndRecover(
+        address from,
+        address to,
+        uint256 gameId,
+        uint256[] calldata assetIds,
+        uint256[] calldata values
+    ) external override gameManagerOnly() {
+        _destroyGame(from, to, gameId);
+        _recoverAssets(from, to, gameId, assetIds, values);
+    }
+
+    /// @notice Function to burn a GAME token
+    /// @param from The address of the one destroying the game
+    /// @param to The address to send all GAME assets to
+    /// @param gameId The id of the GAME to destroy
+    function destroyGame(
+        address from,
+        address to,
+        uint256 gameId
+    ) external override gameManagerOnly() {
+        _destroyGame(from, to, gameId);
+    }
+
+    /// @notice Function to add assets to an existing GAME
+    /// @param from The address of the one creating the game (may be different from msg.sender if metaTx)
+    /// @param gameId The id of the GAME to add asset to
+    /// @param assetIds The id of the asset to add to GAME
+    /// @param values The amount of each asset to add to GAME
+    /// @param uri The new uri to set
+    function addAssets(
+        address from,
+        uint256 gameId,
+        uint256[] memory assetIds,
+        uint256[] memory values,
+        string memory uri
+    ) public override gameManagerOnly() {
+        require(assetIds.length == values.length && assetIds.length != 0, "INVALID_INPUT_LENGTHS");
+        for (uint256 i = 0; i < assetIds.length; i++) {
+            _gameAssets[gameId][assetIds[i]] = values[i];
```

Comment:
if you add to existing assetId, it should add, not replace the value

Reply:
indeed. I will strangthen test coverage first as this should currently be failing. 
Then I'll fix to add instead.

Reply:
I think an improvvement here would involve 1 of 2 approaches:
a: I expose _gameAssets, either by making `mapping(uint256 => mapping(uint256 => uint256)) private _gameAssets;` public, or adding an external getter for it. This lets others query the data, but also makes it easy to test the contract's state.
B: Look at using smock to mock this with an external function wrapper I can test against.
I prefer A, but if you don't want to expose the data I can go with B. (really, the data is all public anyway, A would just make it convenient)